### PR TITLE
Add documentation for sidecars

### PR DESCRIPTION
**Pull request by xeviknal**
Adding documentation to all present sidecar validations. It includes documentation for the latest PR opened for that: https://github.com/kiali/kiali/pull/2500#

List of validations:
https://github.com/kiali/kiali/pull/2500/files#diff-00df504feeb66372dea6b00a33164c66R202-R221

### CODE REVIEW

File: content/documentation/validations/index.adoc [hunk 1]
```diff
@@ -687,10 +687,156 @@ include::/data/files/validation_examples/601.yaml[]
...
+https://github.com/kiali/kiali/tree/master/business/checkers/sidecars/workload_selector_checker.go[Validator source code, window="_blank"] +
+https://istio.io/docs/reference/config/networking/sidecar/[Sidecar documentation, window="_blank"] +
+
+'''
+
+=== KIA1002 - More than one selector-less Sidecar in the same namespace
+
+The Sidecar resources are used for configuring the sidecar proxies in the service mesh. This resource is aimed to target either specific workloads, all workloads in a namespace or all the sidecars of the service mesh. Each Sidecar resource living in a namespace (except the one living in the control plane namespace) effects only to the workloads in the same namespace.
+
+Selector-less sidecars are those sidecars that doesn't have the `workloadSelector` field and thus they are pointing to all workloads running in the same namespace.
```
Comment (lucasponce) [inline content/documentation/validations/index.adoc]: "that doesn't" -> "that don't" I guess you are talking in plural in this sentence.

File: content/documentation/validations/index.adoc [hunk 2]
```diff
@@ -687,10 +687,156 @@ include::/data/files/validation_examples/601.yaml[]
...
+https://github.com/kiali/kiali/tree/master/business/checkers/sidecars/multi_match_checker.go[Validator source code, window="_blank"] +
+https://istio.io/docs/reference/config/networking/sidecar/[Sidecar documentation, window="_blank"] +
+
+'''
+
+=== KIA1003 - Invalid host format. 'namespace/dnsName' format expected
+
+The Sidecar resources are used for configuring the sidecar proxies in the service mesh. IstioEgressListener specifies the properties of an outbound traffic listener on the sidecar proxy attached to a workload instance.
+
+In the host field, there is the list of hosts exposed to the workload. Each host in the list must have the `namespace/dnsName` format.
```
Comment (lucasponce) [inline content/documentation/validations/index.adoc]: "In the host field, it is the list of host exposed" ?
Not sure if this sounds better, anyway not a blocker.

File: content/documentation/validations/index.adoc [hunk 3]
```diff
@@ -687,10 +687,156 @@ include::/data/files/validation_examples/601.yaml[]
...
+==== See Also
+
+https://github.com/kiali/kiali/tree/master/business/checkers/sidecars/multi_match_checker.go[Validator source code, window="_blank"] +
+https://istio.io/docs/reference/config/networking/sidecar[Sidecar documentation: first warning, window="_blank"] +
+
 [#virtualservices]
 == VirtualServices
 
-=== KIA1001 - DestinationWeight on route doesn't have a valid service (host not found)
+=== KIA1101 - DestinationWeight on route doesn't have a valid service (host not found)
```
Comment (lucasponce) [inline content/documentation/validations/index.adoc]: I guess these codes were the last one we discussed, right ?
We can't change the code for errors.

File: content/documentation/validations/index.adoc (hunk 2 above)
Comment (xeviknal) [reply content/documentation/validations/index.adoc]: > The hosts list is the list of hosts that will be exposed to the workload.

Better?

File: content/documentation/validations/index.adoc (hunk 3 above)
Comment (xeviknal) [reply content/documentation/validations/index.adoc]: Yes, they are the ones we discussed.

**Comment by xeviknal** [comment]
Comments approached :)

**Pull request merged**